### PR TITLE
Handle add member request in the coordinator

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

This PR adds necessary functionalities to mange a topology mangement requests from client (gateway) to server (coordinator broker). 
- Defines the API, and protobuf messages.
- Adds server for handling requests for topology Manager. `TopologyRequestServer` will be started by the ClusterTopologyRequestService in the coordinator.
- Add `TopologyManagementRequestSender` that sends requests to the server running in the coordinator. The gateway can use this to interact with the coordinator.
- Add support for handling request to add new members.

Note that the api might change as we are still discussing the user facing api - such as whether the add member request should take one member id or a list of member ids. We can easily change this later.

## Related issues

related #14462 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: topology/src/main/java/io/camunda/zeebe/topology/api/TopologyManagementAPI.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.topology.api;
+
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.topology.api.TopologyManagementRequests.AddMembersRequest;
+import io.camunda.zeebe.topology.api.TopologyManagementResponses.TopologyChangeStatus;
+
+/** Defines the API for the topology management requests. */
+public interface TopologyManagementAPI {
```
Comment (oleschoenburg) [inline topology/src/main/java/io/camunda/zeebe/topology/api/TopologyManagementAPI.java:15]: :wrench: Should be `TopologyManagementApi`. Same for the test class and some local variables.

File: topology/src/main/java/io/camunda/zeebe/topology/api/TopologyManagementRequests.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.topology.api;
+
+import io.atomix.cluster.MemberId;
+import java.util.Set;
+
+/** Defines the supported requests for the topology management. */
+public final class TopologyManagementRequests {
```
Comment (oleschoenburg) [inline topology/src/main/java/io/camunda/zeebe/topology/api/TopologyManagementRequests.java:14]: :wrench: IMO this is not needed, we can just use the types defined in the `.proto` directly.

File: topology/src/main/java/io/camunda/zeebe/topology/api/TopologyManagementResponses.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.topology.api;
+
+public final class TopologyManagementResponses {
```
Comment (oleschoenburg) [inline topology/src/main/java/io/camunda/zeebe/topology/api/TopologyManagementResponses.java:10]: :wrench: Also not needed, we can use the protobuf types directly.

**Review by lenaschoenburg — APPROVED** [review]
I have some general comments about our usage of protobuf types:
As is, you defined an intermediate representation for request/response types. IMO this is not needed and we can just directly use the types generated by the protobuf schema. This would remove a lot of mapping code with no(?) drawback.

The same applies to `ClusterTopologyGossipState` I think, unless I'm forgetting a good reason not do this there? :thinking:

**Comment by deepthidevaki** [comment]
> As is, you defined an intermediate representation for request/response types. IMO this is not needed and we can just directly use the types generated by the protobuf schema. This would remove a lot of mapping code with no(?) drawback.

This is mostly for seperation of concerns. The handlers of requests shouldn't care that this is serialized using protobuf. Ideally, we should be able to switch to sbe (for example) easily by only switching the serializer without out touching any of the main code processing the requests. Same on the gateway - gateway should be able to use the `TopologyManagementRequestSender` without knowing what serialization is used underneath.
Alternate is to define a wrapper class that wraps the protobuf object, so we don't have to map back and forth from protobuf objects (less copying may be?). But I would rather not use the protobuf objects directly in the `TopologyManagementApi`.

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**